src/ctia/http/routes/version.clj

```diff
@@ -1,7 +1,17 @@
 (ns ctia.http.routes.version
-  (:require [compojure.api.sweet :refer :all]
-            [ring.util.http-response :refer :all]
-            [ctia.schemas.common :refer [VersionInfo]]))
+  (:require [clojure.java
+             [io :as io]
+             [shell :as shell]]
+            [clojure.string :as st]
+            [ctia.schemas.common :refer [VersionInfo]]
+            [compojure.api.sweet :refer :all]
+            [ring.util.http-response :refer :all]))
+
+(defn current-build []
+  (if-let [built-version (io/resource "ctia-version.txt")]
+    built-version
+    (str (:out (shell/sh "git" "log" "-n" "1" "--pretty=format:%H "))
+         (:out (shell/sh "git" "symbolic-ref" "--short" "HEAD")))))
```

Comment:
avoiding running code at compile time is a rule thumb to save you much pain. .(that `def` would be running code on load 👎 )

Why not just:

```clj
(def current-build
  (memoize (fn []
             (if-let [built-version (io/resource "ctia-version.txt")]
               built-version
               (str (:out (shell/sh "git" "log" "-n" "1" "--pretty=format:%H "))
                    (:out (shell/sh "git" "symbolic-ref" "--short" "HEAD")))))))
```